nanoFramework.Hardware.Esp32.Rmt/RmtChannelSettings.cs

```diff
@@ -0,0 +1,142 @@
+﻿//
+// Copyright (c) .NET Foundation and Contributors
+// See LICENSE file in the project root for full license information.
+//
+
+using System;
+
+namespace nanoFramework.Hardware.Esp32.Rmt
+{
+    /// <summary>
+    /// Base class for shared RMT Channel settings.
+    /// </summary>
+    public abstract class RmtChannelSettings
+    {
+        private int _channel;
+        private int _pinNumber;
+        private byte _clockDivider;
+        private byte _numberOfMemoryBlocks;
+        private int _bufferSize;
+        private bool _signalInverterEnabled;
+
+        /// <summary>
+        /// Gets or sets the channel number.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Cannot make changes to the property after the RMT channel is initialized.
+        /// Use the methods on the channel instance to make configuration changes.</exception>
+        public int Channel
+        {
+            get => _channel;
+            set => _channel = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the GPIO pin number to be used with the specified channel.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Cannot make changes to the property after the RMT channel is initialized.
+        /// Use the methods on the channel instance to make configuration changes.</exception>
+        public int PinNumber
+        {
+            get => _pinNumber;
+            set => _pinNumber = value;
```

Comment:
is <0 valid? And is this checked on the native side?

Reply:
Good point, no it is not valid. I've added a check on the managed side. On the native side, however, I'm relying on IDF to throw an error if such invalid value makes it there.

Reply:
No need to have duplicate checks! If this is checked in native, no need to add anything here. 
We just need to be sure that the check failure is properly reported to the developer.

Reply:
Understood. If that's the case, perfect them. These make sense and are not taking up space nor are repeated validation. Let's have them!